analyticsShared/src/commonMain/kotlin/mealz/ai/AbstractSharedAnalytics.kt

```diff
@@ -24,19 +24,43 @@ abstract class AbstractSharedAnalytics {
 
     abstract fun sendRequest(event: PlausibleEvent);
 
+    private fun validatePath(path: String) {
+        // If the path does not contain "/miam/", it may just be the URL of the page (web) or
+        // an empty path (mobile) so we assume it is valid
+        if (!path.contains("/miam/")) return
+
+        val validParts = "||miam|recipes|liked|categories|my-meals|detail|replace-item|sponsor|meals-planner|catalog|results|basket-preview|finalize|"
+        var pathWithoutURL = path.substringAfter("/miam/")
+        // Not using a list is the lightest way to do it when compiled in JS
+        while (pathWithoutURL.isNotEmpty()) {
+            // Find the index of the next '/' to separate path segments
+            val nextSlashIndex = pathWithoutURL.indexOf('/').takeIf { it != -1 } ?: pathWithoutURL.length
+            val part = pathWithoutURL.substring(0, nextSlashIndex)
+            // If the part is not in the valid set of path segments, throw an exception
+            if (!validParts.contains("|$part|")) {
+                throw IllegalArgumentException("Invalid path : $part")
```

Comment:
should we throw error or just remove it?

Reply:
I think removing it would be a pb, this could cause some paths like /miam/recipes/my-meal/detail to change into /miam/recipes/detail